src/workerd/jsg/jsg.c++

```diff
@@ -270,6 +271,11 @@ void Lock::terminateExecution() {
   v8Isolate->TerminateExecution();
 }
 
+void Lock::pumpMessageLoop() {
+  auto platform = IsolateBase::from(v8Isolate).getDefaultPlatform();
+  while (v8::platform::PumpMessageLoop(platform, v8Isolate, v8::platform::MessageLoopBehavior::kDoNotWait)) {}
```

Comment:
Oy, it feels gnarly to add a while loop here. I'm assuming that we are assured enough that this won't go into an infinite loop at all? A comment here would probably be helpful.

Reply:
Will add a comment - it is (theoretically) possible to keep adding FinalizationRegistry cleanup tasks from inside the cleanup callback. But, the while loop does run under cpu limiter, so it will get terminated at some point.